asapdiscovery-simulation/asapdiscovery/simulation/schema/fec.py

```diff
@@ -0,0 +1,387 @@
+        return protocol_class(settings=protocol_settings)
+
+
+class FreeEnergyCalculationNetwork(_FreeEnergyBase):
+    """
+    A schema of a FEC network created by the FreeEnergyCalculationFactory which contains all runtime settings and can
+    be converted to local openFE inputs or submitted to alchemiscale.
+    """
+
+    type: Literal["FreeEnergyCalculationNetwork"] = "FreeEnergyCalculationNetwork"
+    dataset_name: str = Field(
+        ...,
+        description="The name of the dataset, this will be used for local files and the alchemiscale network.",
+    )
+    network: PlannedNetwork = Field(
+        ...,
+        description="The planned free energy network with atom mappings between ligands.",
+    )
+    receptor: str = Field(
+        ...,
+        description="The JSON str of the receptor which should be used in the FEC calculation.",
+    )
+    results: Optional[AlchemiscaleResults] = Field(
+        None,
+        description="The results object which tracks how the calculation was run locally or on alchemiscale and stores the physical results.",
+    )
+
+    class Config:
+        """Overwrite the class config to freeze the results model"""
+
+        allow_mutation = False
+
+    def to_openfe_receptor(self) -> openfe.ProteinComponent:
+        return openfe.ProteinComponent.from_json(self.receptor)
+
+    def to_alchemical_network(self) -> openfe.AlchemicalNetwork:
+        """
+        Create an openfe AlchemicalNetwork from the planned network which can be submitted to alchemiscale or ran locally
+
+        Returns:
+            An openfe.AlchemicalNetwork created from the schema.
+        """
+        transformations = []
+        # do all openfe conversions
+        ligand_network = self.network.to_ligand_network()
+        solvent = self.solvent_settings.to_solvent_component()
+        receptor = self.to_openfe_receptor()
+        protocol = self.to_openfe_protocol()
+
+        # build the network
+        for mapping in ligand_network.edges:
+            for leg in ["solvent", "complex"]:
+                sys_a_dict = {"ligand": mapping.componentA, "solvent": solvent}
+                sys_b_dict = {"ligand": mapping.componentB, "solvent": solvent}
+
+                if leg == "complex":
+                    sys_a_dict["protein"] = receptor
+                    sys_b_dict["protein"] = receptor
+
+                system_a = openfe.ChemicalSystem(
+                    sys_a_dict, name=f"{mapping.componentA.name}_{leg}"
+                )
+                system_b = openfe.ChemicalSystem(
+                    sys_b_dict, name=f"{mapping.componentB.name}_{leg}"
+                )
+
+                transformation = openfe.Transformation(
+                    stateA=system_a,
+                    stateB=system_b,
+                    mapping={"ligand": mapping},
+                    protocol=protocol,  # use protocol created above
+                    name=f"{self.dataset_name}{system_a.name}_{system_b.name}",
```

Comment:
Consider a separator between `{self.dataset_name}` and `{system_a.name}`.

Reply:
Followup issue #465